### PR TITLE
Add org.kde.kasts

### PR DESCRIPTION
**Pull request by espidev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://invent.kde.org/plasma-mobile/kasts (I obtained permission from the contributors over matrix)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). 
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by espidev** [comment]
bot, build org.kde.kasts

**Comment by espidev** [comment]
bot, build org.kde.kasts

File: org.kde.kasts.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "command": "kasts",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri"
+    ],
```
Comment (TobiasFella) [inline org.kde.kasts.json:14]: We also need permission to use pulseaudio and the session bus for MPRIS

**Review by TobiasFella — CHANGES_REQUESTED** [review]

File: org.kde.kasts.json (hunk 1 above)
Comment (espidev) [reply org.kde.kasts.json:14]: :man_facepalming:

**Comment by espidev** [comment]
bot, build org.kde.kasts

File: org.kde.kasts.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        "--socket=wayland",
+        "--device=dri",
+        "--socket=pulseaudio",
+        "--own-name=org.mpris.MediaPlayer2.kasts"
+    ],
+    "modules": [
+        {
+            "name": "kasts",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [ "-DBUILD_TESTING=OFF", "-DCMAKE_BUILD_TYPE=RelWithDebInfo" ],
```
Comment (TobiasFella) [inline org.kde.kasts.json:21]: As far as i can tell flathub doesn't usually do RelWithDebInfo builds

Also, we don't have tests, so disabling them is not really necessary

**Review by TobiasFella — CHANGES_REQUESTED** [review]

**Comment by espidev** [comment]
bot, build org.kde.kasts

**Comment by TobiasFella** [comment]
LGTM

File: 0001-add-url.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --git a/org.kde.kasts.appdata.xml b/org.kde.kasts.appdata.xml
+index 4736b7f..c44b0bf 100644
+--- a/org.kde.kasts.appdata.xml
++++ b/org.kde.kasts.appdata.xml
+@@ -42,6 +42,7 @@
+   <metadata_license>CC0-1.0</metadata_license>
+   <project_license>GPL-2.0-or-later</project_license>
+   <url type="bugtracker">https://invent.kde.org/plasma-mobile/kasts/-/issues</url>
++  <url type="homepage">https://invent.kde.org/plasma-mobile/kasts</url>
```
Comment (ognarb) [inline 0001-add-url.patch:9]: This should probably be https://apps.kde.org/kasts/ instead

**Review by ognarb — CHANGES_REQUESTED** [review]

**Comment by espidev** [comment]
bot, build org.kde.kasts

**Comment by mogwa1** [comment]
Just tested it (a bit late to the party); looks good!

**Comment by nedrichards** [comment]
@flathub/kde

**Comment by tsdgeos** [comment]
looks good to me

**Comment by barthalion** [comment]
I'm not exactly sure how it looks like with Breeze but it seems to be missing icons on GNOME 40 here.

image

**Comment by espidev** [comment]
> I'm not exactly sure how it looks like with Breeze but it seems to be missing icons on GNOME 40 here.
> 
> image

Seems those icons also don't exist in Adwaita, perhaps I should force breeze theme? :/

**Comment by espidev** [comment]
bot, build org.kde.kasts

**Comment by espidev** [comment]
bot, build org.kde.kasts

**Comment by espidev** [comment]
> I'm not exactly sure how it looks like with Breeze but it seems to be missing icons on GNOME 40 here.
> 
> image

Try again?

**Comment by barthalion** [comment]
LGTM

/merge

**Pull request closed without merging**